### PR TITLE
test: add unit tests for api.py (TrueNASAPI, 100% coverage)

### PR DESCRIPTION
**Pull request by kayl-codes**
## Proposed change
Adds a unit test suite for `api.py` (`TrueNASAPI`), reaching 100% statement coverage. Unlike `config_flow.py`/`coordinator.py`, `TrueNASAPI` has no Home Assistant dependency at all — it only wraps `aiotruenas.TrueNASClient` — so it imports and instantiates as a real package module, with the underlying `aiotruenas` client mocked so no network I/O happens.

Covers `_summarize_payload`, `_classify_exception`, and every `TrueNASAPI` method/property (`connect`, `disconnect`/`close`, `connected`, `connection_test`, `query`, `error`/`scheme` properties), including exception-mapping and debug-logging branches.

## Type of change
- [x] Code quality improvements to existing code or addition of tests

## Additional information
No production code changes — test-only PR. This completes the four-module test-suite roadmap (apiparser.py #30, config_flow.py #31, coordinator.py #32, api.py — this PR).

## Checklist
- [x] The code change is tested and works locally.
- [x] The code has been formatted and linted using Ruff.
- [x] Tests have been added to verify that the new code works.
- [ ] Documentation added/updated if required.
- [x] Tested against the latest Home Assistant version.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by Sourcery

Add a comprehensive unit test suite for the TrueNASAPI wrapper in api.py, achieving full coverage without changing production code.

Tests:
- Introduce unit tests for TrueNASAPI methods, properties, and helpers, including connection handling, error classification, and payload summarization.
- Verify debug logging behavior and error propagation for TrueNASAPI queries using mocked aiotruenas clients.

### CODE REVIEW

File: tests/test_api.py [hunk 1]
```diff
@@ -0,0 +1,299 @@
...
+def test_summarize_payload_lists_shape() -> None:
+    assert _summarize_payload([1, 2, 3]).startswith("list[3]")
+
+
+def test_summarize_payload_dict_shape() -> None:
+    assert _summarize_payload({"a": 1, "b": 2}).startswith("dict[2 keys]")
+
+
+def test_summarize_payload_other_type_shape() -> None:
+    assert _summarize_payload("hello").startswith("str ")
```
[the remark this answers is not in the document]
Comment (kayl-codes) [reply tests/test_api.py:96]: Done in 2415766: added edge-case tests for the empty list (`list[0]`), empty dict (`dict[0 keys]`), and `None`. Note that `None` summarizes as `NoneType None` (shape comes from `type(data).__name__`), so the test asserts that exact string rather than a `None` prefix.

File: tests/test_api.py [hunk 2]
```diff
@@ -0,0 +1,299 @@
...
+
+async def test_query_call_error_falls_back_to_str_then_unknown() -> None:
+    api = _api_with_mock_client()
+    api._client.connected = True
+    api._client.call = AsyncMock(side_effect=TrueNASCallError("boom"))
+    assert await api.query("system.info") is None
+    assert api.error == "boom"
+
+
+async def test_query_other_truenas_error_classifies_during_call() -> None:
```
[the remark this answers is not in the document]
Comment (kayl-codes) [reply tests/test_api.py]: Done in 2415766: added `test_query_propagates_non_truenas_exceptions`. One correction to the suggestion: `query()` does **not** fall back to `ERR_UNKNOWN` for non-TrueNAS exceptions — it only catches `TrueNASError` subclasses, so a generic exception propagates to the caller. That is deliberate (aiotruenas only raises `TrueNASError` subclasses, so anything else is a bug that should surface; the coordinator wraps each poll job defensively), and the new test documents exactly that propagation behavior with `pytest.raises`.

File: tests/test_api.py [hunk 3]
```diff
@@ -0,0 +1,299 @@
...
+async def test_connection_test_fails_when_query_returns_none() -> None:
+    api = _api_with_mock_client()
+    api._client.connected = True
+    api._client.call = AsyncMock(return_value=None)
+    ok, error = await api.connection_test()
+    assert ok is False
+    assert error == ERR_MALFORMED_RESULT
+
+
+async def test_connection_test_succeeds() -> None:
```
[the remark this answers is not in the document]
Comment (kayl-codes) [reply tests/test_api.py]: Done in 2415766: added `test_connection_test_connects_before_querying` (starts disconnected, `connect()` succeeds and flips the client to connected, then the `system.info` query runs — asserting both the awaited connect and the awaited call) and `test_connection_test_fails_when_connect_raises` (connect raises `TrueNASConnectionRefusedError` and `connection_test` returns `(False, ERR_CONNECTION_REFUSED)`). Also addressed the overall comments in the same commit: the module-level helper is now a set of pytest fixtures (`api`/`connected_api`/`closed_api`), and the closed state is reached via the public `close()` instead of poking `_closed`.

**Comment by kayl-codes** [comment]
@sourcery-ai review
